cdap-cli/src/main/java/co/cask/cdap/cli/command/StartDebugProgramCommand.java

```diff
@@ -21,57 +21,16 @@
 import co.cask.cdap.cli.ElementType;
 import co.cask.cdap.cli.english.Article;
 import co.cask.cdap.cli.english.Fragment;
-import co.cask.cdap.cli.exception.CommandInputError;
-import co.cask.cdap.cli.util.AbstractAuthCommand;
-import co.cask.cdap.cli.util.ArgumentParser;
 import co.cask.cdap.client.ProgramClient;
-import co.cask.common.cli.Arguments;
-import com.google.gson.Gson;
-
-import java.io.PrintStream;
-import java.util.Map;
 
 /**
  * Starts a program in debug mode.
  */
-public class StartDebugProgramCommand extends AbstractAuthCommand {
-
-  private static final Gson GSON = new Gson();
-
-  private final ProgramClient programClient;
-  private final ElementType elementType;
+public class StartDebugProgramCommand extends StartProgramCommand {
 
   public StartDebugProgramCommand(ElementType elementType, ProgramClient programClient, CLIConfig cliConfig) {
-    super(cliConfig);
-    this.elementType = elementType;
-    this.programClient = programClient;
-  }
-
-  @Override
-  public void perform(Arguments arguments, PrintStream output) throws Exception {
-    String[] programIdParts = arguments.get(elementType.getArgumentName().toString()).split("\\.");
-    if (programIdParts.length < 2) {
-      throw new CommandInputError(this);
-    }
-
-    String appId = programIdParts[0];
-    String programId = programIdParts[1];
-
-    String runtimeArgsString = arguments.get(ArgumentName.RUNTIME_ARGS.toString(), "");
-    if (runtimeArgsString == null || runtimeArgsString.isEmpty()) {
-      // run with stored runtime args
-      programClient.start(appId, elementType.getProgramType(), programId, true);
-      runtimeArgsString = GSON.toJson(programClient.getRuntimeArgs(appId, elementType.getProgramType(), programId));
-      output.printf("Successfully started %s '%s' of application '%s' with stored runtime arguments '%s'\n",
-                    elementType.getTitleName(), programId, appId, runtimeArgsString);
-    } else {
-      // run with user-provided runtime args
-      Map<String, String> runtimeArgs = ArgumentParser.parseMap(runtimeArgsString);
-      programClient.start(appId, elementType.getProgramType(), programId, true, runtimeArgs);
-      output.printf("Successfully started %s '%s' of application '%s' with provided runtime arguments '%s'\n",
-                    elementType.getTitleName(), programId, appId, runtimeArgsString);
-    }
-
+    super(elementType, programClient, cliConfig);
+    this.isDebug = true;
```

Comment:
I just didn't think that's necessary, because then it would be possible to directly instantiate the parent class with either value (true or false), which would then be inconsistent with the `getPattern` and `getDescription` of that class.

Alternative is just to merge the two patterns/descriptions into one class, if that's what you mean?